src/JuMP.jl

```diff
@@ -56,51 +61,82 @@ export
 include("JuMPContainer.jl")
 include("utils.jl")
 
+const MOIVAR = MOI.VariableReference
+const MOICON{F,S} = MOI.ConstraintReference{F,S}
+const LBREF = MOICON{MOI.SingleVariable,MOI.GreaterThan{Float64}}
+const UBREF = MOICON{MOI.SingleVariable,MOI.LessThan{Float64}}
+const FIXREF = MOICON{MOI.SingleVariable,MOI.EqualTo{Float64}}
+const INTREF = MOICON{MOI.SingleVariable,MOI.Integer}
+const BINREF = MOICON{MOI.SingleVariable,MOI.ZeroOne}
+
+@MOIU.instance JuMPInstance (ZeroOne, Integer) (EqualTo, GreaterThan, LessThan, Interval) (Zeros, Nonnegatives, Nonpositives) () (SingleVariable,) (ScalarAffineFunction,) (VectorOfVariables,) (VectorAffineFunction,)
+
+# dummy solver
+type UnsetSolver <: MathProgBase.AbstractMathProgSolver
+end
+
 ###############################################################################
 # Model class
 # Keeps track of all model and column info
 abstract type AbstractModel end
-type Model <: AbstractModel
-    obj#::QuadExpr
-    objSense::Symbol
+mutable struct Model <: AbstractModel
+
+    instance::JuMPInstance{Float64}
+    # special variablewise properties that we keep track of:
+    # lower bound, upper bound, fixed, integrality, binary
+    variabletolowerbound::Dict{MOIVAR,LBREF}
+    variabletoupperbound::Dict{MOIVAR,UBREF}
+    variabletofix::Dict{MOIVAR,FIXREF}
+    variabletointegrality::Dict{MOIVAR,INTREF}
+    variabletozeroone::Dict{MOIVAR,BINREF}
```

Comment:
`VariableToValueMap` is meant to be user facing and corresponds to MOI attributes. We will return them to users e.g., for multiple solutions, and users can use them to provide attributes like a starting point or starting basis.
`variabletoupperbound` is an internal data structure that will never be visible to users. I guess it would be okay to make it a `VariableToValueMap` but to me it confuses two different concepts.